### ISSUE TITLE
Condition "OutOfDisk" has invalid status Unknown

### ISSUE THREAD

**Issue by afritzler** [gardener/machine-controller-manager#214]
Somehow we are seeing a lot of this kind of errors on OpenStack and Azure lately

```
Node shoot--core--CLUSTER_NODE_WORKER_NAME is unhealthy: condition "OutOfDisk" has invalid status Unknown (expected False) due to NodeStatusNeverUpdated: Kubelet never posted node status.
```

Isn't the machine controller supposed to recreate those machines eventually? Right now it looks like they stay around forever. I know it is dangerous to delete machines when the kubelet is not posting any status anymore - could be that there is a temporary network outage.

**Comment by rfranzke** [comment]
I've seen this only happening for 1.13 clusters. The `OutOfDisk` condition has been removed from K8s. I'm not sure why it appears in the list of conditions again, must be a Kubernetes bug. The VMs themselves are completely healthy (all other conditions are fine and status is `Ready`) - hence, MCM does not recreate them.
We probably should mitigate the issue by removing these conditions out of the `Node` object for 1.13 clusters...

**Comment by PadmaB** [comment]
this bug seems to be fixed now, refer to the issue for details https://github.com/kubernetes/kubernetes/pull/72507

**Comment by rfranzke** [comment]
Thanks @PadmaB for linking this one. I have not found a cherry-pick PR to `release-1.13` branch yet...
Even if it will be created - it will take a while until the new release it cut. I will prepare some implementation in Gardener that purges the unused condition.

Btw, you could also remove it manually yourself by running

```
for node in `k get no -o jsonpath={..metadata.name}`; do index=$(k get no $node -o json | jq -r '.status.conditions | index(.[] | select(.type=="OutOfDisk"))'); if [[ ! -z $index ]]; then curl -k -v -XPATCH  -H "Accept: application/json" -H "Content-Type: application/json-patch+json" "localhost:8001/api/v1/nodes/$node/status" --data "[{\"op\":\"remove\",\"path\":\"/status/conditions/$index\"}]"; fi; done
```

(assuming that you have proxied the API server to `localhost:8001`).

**Comment by afritzler** [comment]
I just cherry picked that puppy for the 1.13 release branch. https://github.com/kubernetes/kubernetes/pull/73394

**Comment by vlerenc** [comment]
Thanks together for the very nice collaboration!

**Comment by prashanth26** [comment]
Do we have anything else pending on this issue? If not, is it safe to close this issue?

**Comment by afritzler** [comment]
Since the cherry pick PR has been approved (fixed in 1.13 onwards) and we also have the mitigation in Gardener itself [0] it is safe to close this issue now.

[0] https://github.com/gardener/gardener/pull/691

**Comment by prashanth26** [comment]
Thank you.
